CLDR-8823 Don't use Jan, Feb, for non-gregorian calendar systems

srl295 opened the pull request:
- update PathHeader with a %G, a list of calendars which use the Gregorian month names.

CLDR-8823

- [X] This PR completes the ticket.

<!--
Thank you for your pull request.
Please see https://cldr.unicode.org/index/process for general
information on contributing to CLDR.

1. Make sure the ticket is filed at
https://unicode-org.atlassian.net/projects/CLDR/
2. Update the PR title and first line of this
message to include the ticket ID (CLDR-_____)
3. You will be automatically asked to sign the contributors’
license before the PR is accepted.
- sign: https://cla-assistant.io/unicode-org/cldr
- license: https://www.unicode.org/copyright.html#License
-->

<img width="525" alt="image" src="https://github.com/unicode-org/cldr/assets/855219/439eb83b-b092-41ab-a0ef-661dc885dfc9">

Includes a test. The test catches some issues.

ALLOW_MANY_COMMITS=true

pedberg-icu commented:
Changes look good so far but there is a test failure...

AEApple commented:
Does this mean it will show the month names of the default calendar for that locale?

pedberg-icu commented:
> Does this mean it will show the month names of the default calendar for that locale?

This is about the key in the left column (not the English name, which will remain as is). Instead of the key showing Jan, Feb, Mar... for non-gregorian calendars, it will show M1, M2, M3....

AEApple commented:
> > Does this mean it will show the month names of the default calendar for that locale?
> 
> This is about the key in the left column (not the English name, which will remain as is). Instead of the key showing Jan, Feb, Mar... for non-gregorian calendars, it will show M1, M2, M3....

Okay in the ticket the "English" showed "Month 1", "Month 2" etc.

macchiati commented:
I presume that this does not change Gregorian -based calendars, right? And
by that I include Julian, Thai Buddhist, etc

On Wed, Jul 10, 2024, 21:22 Annemarie Apple ***@***.***>
wrote:

> Does this mean it will show the month names of the default calendar for
> that locale?
>
> This is about the key in the left column (not the English name, which will
> remain as is). Instead of the key showing Jan, Feb, Mar... for
> non-gregorian calendars, it will show M1, M2, M3....
>
> Okay in the ticket the "English" showed "Month 1", "Month 2" etc.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/unicode-org/cldr/pull/3861#issuecomment-2221996825>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACJLEMC7OWALPF36KJ42FETZLYB67AVCNFSM6AAAAABKVIK5LOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDEMRRHE4TMOBSGU>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>

macchiati commented on tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/PathHeader.txt:
```diff
@@ -20,7 +20,11 @@
-%O = (gregorian|chinese|coptic|dangi|ethiopic|hebrew|indian|islamic|persian)
+# calendar systems that should use Gregorian months (just Gregorian)
+# We don't list roc, etc here because their months are hidden.
+%G = (gregorian)
```
This should support not only gregorian, but also gregorian-based calendars. (Those that use the same 12 month solar calendar: Thai Buddhist, Julian, etc.

macchiati replied:
Here are the calendars. Unfortunately, we don't have data here as to which is gregorian-based.
```
            <type name="buddhist" description="Thai Buddhist calendar"/>
            <type name="chinese" description="Traditional Chinese calendar"/>
            <type name="coptic" description="Coptic calendar"/>
            <type name="dangi" description="Traditional Korean calendar" since="22.1"/>
            <type name="ethioaa" description="Ethiopic calendar, Amete Alem (epoch approx. 5493 B.C.E)" alias="ethiopic-amete-alem"/>
            <type name="ethiopic" description="Ethiopic calendar, Amete Mihret (epoch approx, 8 C.E.)"/>
            <type name="gregory" description="Gregorian calendar" alias="gregorian"/>
            <type name="hebrew" description="Traditional Hebrew calendar"/>
            <type name="indian" description="Indian calendar"/>
            <type name="islamic" description="Hijri calendar"/>
            <type name="islamic-umalqura" description="Hijri calendar, Umm al-Qura" since="24"/>
            <type name="islamic-tbla" description="Hijri calendar, tabular (intercalary years [2,5,7,10,13,16,18,21,24,26,29] - astronomical epoch)" since="24"/>
            <type name="islamic-civil" description="Hijri calendar, tabular (intercalary years [2,5,7,10,13,16,18,21,24,26,29] - civil epoch)" since="24"/>
            <type name="islamic-rgsa" description="Hijri calendar, Saudi Arabia sighting" since="24"/>
            <type name="iso8601" description="ISO calendar (Gregorian calendar using the ISO 8601 calendar week rules)" since="2.0"/>
            <type name="japanese" description="Japanese Imperial calendar"/>
            <type name="persian" description="Persian calendar"/>
            <type name="roc" description="Republic of China calendar"/>
            <type name="islamicc" description="Civil (algorithmic) Arabic calendar" deprecated="true" preferred="islamic-civil"  alias="islamic-civil"/>
```

macchiati replied:
I ran a quick check, and it appears that the following are gregorian-based: 
[gregoriy, iso8601, buddhist, japanese, roc]

Also filed a ticket to add metadata to cldr, so that this can be unhardcoded

srl295 replied:
So actually it seems the "intent" was to hide those so that you don't have a Japanese month January with a different translation than Gregorian January. That mystified me too but I think that's why %G is just one and not the others

macchiati replied:
@pedberg-icu should also weigh in on this.

- It clearly should not be limited to exactly 'gregory', since iso8601 uses the same months, etc.
- In the case of thai buddhist, japanese, roc, M2, M2 etc seem ok.

srl295 replied:
If it's here, then it's visible for editing.  Do we want users giving a different translation for January in 8601 than for gregorian?

srl295 replied:
@macchiati Specifically, this PR does not make any changes to the visibility (HIDE) or textual content for these calendars: `iso8601, buddhist, japanese, roc`. 

All that it does is change the following calendars to no longer have month codes based on Gregorian:

`chinese|coptic|dangi|ethiopic|hebrew|indian|islamic|persian`

srl295 commented:
OK the test failure was because I dropped Chinese inadvertently. I was just trying to subdivide %O into %G + %O, with different text.

The err message was accurate but didn't quite give me enough context. I updated it.

srl295 commented:
Maybe we should discuss in a future meeting.

macchiati commented on tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/PathHeader.txt:
```diff
@@ -20,7 +20,11 @@
+# We don't list roc, etc here because their months are hidden.
+%G = (gregorian)
+# all others use M## form months.
+%O = (chinese|coptic|dangi|ethiopic|hebrew|indian|islamic|persian)
```
This isn't working, I think, because what we want is that every calendar falls into one of these groups.

- Certain calendars are hidden.
- Certain calendars are non-gregorian-based, and use M1
- Certain calendars are gregorian-based, and use Jan

Because we NEVER want to miss a calendar, exactly one of these groups should be "any", and that needs to come last. That should not be the Hidden one, because we want to be safe, and better to show a calendar that we shouldn't, than hide a calendar that should be shown.

So it should be:

So
# all the calendars that should be hidden

...%XXX...

# all the calendars that should use M1

...%O...

# all the calendars that should use Jan

...%A...

srl295 replied:
Can you detail what "isn't working"?

srl295 commented:
@macchiati  a concrete question.

Currently, (with OR without this PR), Survey Tool does not allow the user to translate "January" differently in the Japanese calendar than in the Gregorian.  

Are you saying that users should be able to change the translation of "January" for the Japanese calendar?

macchiati commented:
Peter would know better, but I believe the same months are used in the
Gregorian Japanese calendar and the Imperial Japanese calendar. The only
difference is the era and year number.

On Mon, Jul 15, 2024, 12:34 Steven R. Loomis ***@***.***>
wrote:

> @macchiati <https://github.com/macchiati> a concrete question.
>
> Currently, (with OR without this PR), Survey Tool does not allow the user
> to translate "January" differently in the Japanese calendar than in the
> Gregorian.
>
> Are you saying that users should be able to change the translation of
> "January" for the Japanese calendar?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/unicode-org/cldr/pull/3861#issuecomment-2229238398>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACJLEMHZMA27SEMCRGNECGLZMQP3FAVCNFSM6AAAAABKVIK5LOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDEMRZGIZTQMZZHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
